main_2024.py

```diff
@@ -255,6 +276,130 @@ def main() -> int:
                 print("geolocation confidence: " + str(detection_world.confidence))
                 print("")
 
+        # Check if all workers are alive in each manager.
+        # If a worker is dead, log the error and restart the worker.
+        # Steps: Terminate and join the worker, fill and drain the
+        # related queues, and create and start a new worker.
+        for manager in range(number_of_managers):
+            if not managers_array[manager].are_workers_alive():
+                managers_array[manager].terminate_workers()
+                managers_array[manager].join_workers()
+                managers_array[manager] = None
+
+                if manager == ManagerType.VIDEO_INPUT.value:
+                    main_logger.error(
+                        "Video Input Worker is dead, attempting to restart.",
+                        frame,
+                    )
+                    video_input_to_detect_target_queue.fill_and_drain_queue()
```

Comment:
I don't think we want to fill the queue before draining, filling the queue with sentinel (None) values will take down the worker reading from that queue, so drain is ok, we also don't necessarily have to drain the queue since a worker that died wouldn't have put bad data into the queue, and draining the queue also blocks the other workers after it until the dead worker is back up

I'm open to suggestions on this one though, since I'm not 100% sure on how to handle queues of dead workers

Reply:
I agree that we don't need to fill the queue before draining. In terms of draining, maybe we should just be draining the preceding queue to ensure that the preceding queue data wasn't what caused the worker to fail. As for draining the succeeding queues, I don't think it is necessary because (as you mentioned) a worker that died wouldn't have put bad data into the queue.

Reply:
that makes sense, I agree